### PR TITLE
Exalt UI Overhaul

### PR DESCRIPTION
**Pull request by Mixelz**
Oh boy another Mixelz UI Overhaul on a Singularity mechanic! **What are the odds!!!**
Changes were implemented to be as unobtrusive as possible, only 2 additional variables for EXALT challenges are required for this.
Ideally we'd also showcase current completion # under the icons like normal challenges & current effect % as well but I am not a good enough programming wizard to implement this. If anyone else wants to take a shot please do so.

## Primary Changes
- EXALT is now a Subtab of Challenges, subtabs are hidden until Exalts are unlocked normally (Sing 25).
  - Above change includes removing EXALT subtab from Singularity Tab and shifting everything required over by one (I hopefully learned my lesson from the Perk/Penalties merge).
- EXALT Rewards are now visually split between **Scaling Rewards** and **Unique Rewards**.
  - Above change includes having an bulletpoint list for all rewards in an exalt (Much easier to read than one run on sentence).
- EXALT Icons now react to the cursor (Cursor Change + Icon Highlight. WOW!!!!1!)
- EXALT Info reorganized for easier reading.

image

### CODE REVIEW

File: src/SingularityChallenges.ts [hunk 1]
```diff
@@ -230,11 +230,34 @@ export class SingularityChallenge {
         this.completions
       )
     }</span></span>
-                <span>${this.rewardDescription}</span>`
+                <span style="color: lightblue">${this.description}</span>`
+  }
+  //Numerates through total reward count for Scaling & Unique string for EXALTS.
+  ScaleString (): string {
```
Comment (KhafraDev) [inline src/SingularityChallenges.ts:236]:
```suggestion
  scaleString (): string {
```

File: src/SingularityChallenges.ts [hunk 2]
```diff
@@ -230,11 +230,34 @@ export class SingularityChallenge {
...
+    for (let i = 1; i <= this.scalingrewardcount; i++) {
+      const list = i18next.t(`singularityChallenge.data.${String(this.HTMLTag)}.ScalingReward${i}`);
+     if (i > this.scalingrewardcount) {}
+    text += i > 1 ? `\n${list}` : list
+    }
+     return text
+  }
+
+  //Ditto. Also worth mentioning this implementation means the list size can be arbitrary!
+  UniqueString (): string {
```
Comment (KhafraDev) [inline src/SingularityChallenges.ts:247]:
```suggestion
  uniqueString (): string {
```

**Review by KhafraDev — CHANGES_REQUESTED** [review]

File: src/SingularityChallenges.ts (hunk 2 above)
Comment (Mixelz) [reply src/SingularityChallenges.ts]: Implmented in follow up commit

File: src/SingularityChallenges.ts (hunk 1 above)
Comment (Mixelz) [reply src/SingularityChallenges.ts]: Implmented in follow up commit

File: src/SingularityChallenges.ts [hunk 3]
```diff
@@ -230,11 +230,34 @@ export class SingularityChallenge {
...
     }</span></span>
-                <span>${this.rewardDescription}</span>`
+                <span style="color: lightblue">${this.description}</span>`
+  }
+  //Numerates through total reward count for Scaling & Unique string for EXALTS.
+  scaleString (): string {
+    let text = ''
+    for (let i = 1; i <= this.scalingrewardcount; i++) {
+      const list = i18next.t(`singularityChallenge.data.${String(this.HTMLTag)}.ScalingReward${i}`);
+     if (i > this.scalingrewardcount) {}
```
Comment (KhafraDev) [inline src/SingularityChallenges.ts:240]: this does nothing

File: src/SingularityChallenges.ts [hunk 4]
```diff
@@ -230,11 +230,34 @@ export class SingularityChallenge {
...
+    }
+     return text
+  }
+
+  //Ditto. Also worth mentioning this implementation means the list size can be arbitrary!
+  uniqueString (): string {
+    let text = ''
+    for (let i = 1; i <= this.uniquerewardcount; i++) {
+      const list = i18next.t(`singularityChallenge.data.${String(this.HTMLTag)}.UniqueReward${i}`);
+     if (i > this.scalingrewardcount) {}
```
Comment (KhafraDev) [inline src/SingularityChallenges.ts:251]: same here

**Review by KhafraDev — CHANGES_REQUESTED** [review]

File: src/SingularityChallenges.ts (hunk 3 above)
Comment (Mixelz) [reply src/SingularityChallenges.ts]: Resolved with commit

File: src/SingularityChallenges.ts (hunk 4 above)
Comment (Mixelz) [reply src/SingularityChallenges.ts]: Resolved with commit

**Pull request merged**